storage/mysql/log_storage_test.go

```diff
@@ -231,7 +231,7 @@ func TestReadWriteTransaction(t *testing.T) {
 			desc:          "unknownBegin",
 			tree:          logTree(-1),
 			wantNeedsInit: true,
-			wantTXRev:     -1,
+			wantTXRev:     0,
```

Comment:
This is surprising, is it hitting `ErrTreeNeedsInit` somehow?

Reply:
Rev = -1 + 1 = 0

Reply:
I had to add a special case for it. Here's the line https://github.com/google/trillian/blob/aa1a68cda895fdf4cf1792f98dbe32f2cccddb23/storage/mysql/log_storage.go#L257

The reason is that, in an uninitialized tree, the first thing we write is an empty signed log root at ... wait for it ...  revision 0. Therefore `writeRevision` for a `ErrTreeNeedsInit` needs to be 0 in order to be consistent.

Reply:
Hmm, I suspect this test has a wrong name - given it's looking for `ErrTreeNeedsInit`, and the comment above talks about unknown trees being handled elsewhere, this test is really testing `uninitializedBegin`, in which case `writeRev==0` is correct :)

Shall we rename the test case?

Reply:
sounds good